File: Mobile/android/wrappers/java/org/Sertainty/VirtualFile.java
package org.Sertainty;

public class VirtualFile {
    static {
        System.loadLibrary(Library.SertaintyCore());
    }

    public VirtualFile(long fileHandle, long vfHandle)
    {
        this.fileHandle = fileHandle;
        this.vfHandle = vfHandle;
    }

    public long read(ByteArray buffer, long mx)
    {
        return read(fileHandle, vfHandle, buffer.getHandle(), mx);
    }

    public void close()
    {
        close(fileHandle, vfHandle);
    }


    private native long read(long fileHandle, long vfHandle, long baBuffer, long mx);
    private native void close(long fileHandle, long vfHandle);

    private long fileHandle;
    private long vfHandle;


}

File: Mobile/ios/swift/VirtualFile.swift
//
//  VirtualFile.swift
//  SampleSwiftUI
//
//  Created by Karim Lalani on 1/21/20.
//  Copyright © 2020 Sertainty. All rights reserved.
//

import Foundation

public class VirtualFile {
    private(set) var handle: OpaquePointer
    private var fileHandle: OpaquePointer?
    
    init(_ file: UxpFile, _ vfHandle: OpaquePointer)
    {
        self.fileHandle = file.handle
        self.handle = vfHandle
    }
    
    func read (_ buffer: ByteArray, _ mx: Int64) -> Int64 {
        return uxpfile_readVirtualFile(fileHandle, handle, buffer.handle, mx)
    }
    
    func close() {
        uxpfile_closeVirtualFile(fileHandle, handle)
    }
}
